### ISSUE TITLE
Problems with Fabric8 and Vanilla kubernetes cluster on AWS

### ISSUE THREAD

**Issue by rpots** [fabric8io/fabric8#6213]
With a Vanilla kubernetes cluster installed on AWS and with the latest verion of fabric8, I am affraid to say that there is a lot of things that simply don't work as expected. This is such a shame for such a promising and exciting project like Fabric8. It seems to me that currently Fabric8 is used in an openshift enviroment mainly, as the Kubernetes+AWS setup I don't read much information about it, nor does the current version works out of the box. The following are the problems I found so far:

1) The publish to docker container does not work. I needed to add the --insecure-registry flag to every docker container.
2) Creating an application on a team, if I leave that page I cannot find that application project anymore. It is like the project is not saved
3) Can't find a way to secure / create users on Farbric8 console
4) Can't connect to the pods terminal on the Fabric8 console (When clicking the terminal icon it fails to open the terminal as says I don't have permissions)
5) Can't access pods logs on Fabric8 console (the log viewer freezes and nothing shows

So far those are the problems I have experienced. I think it would save days of frustration to share this limitations on the Documentation page. If someone helps me, I am available to complete the documentation for Kubernetes + AWS. Did not took the time to dive into the code yet, but so far I am finding it is hard to opt for Fabric8 with Kubernetes on AWS as things just do not work

**Comment by rhuss** [comment]
Thanks for your report, and sorry for your bad experience.  To be honest, as far as I know, we have not used AWS as a target platform yet. All I can say is that Kubernetes is in fact the primary platform we are developing against, and it works smoothly e.g on a local Kubernetes clusters and also on GKE. 

Unfortunately I can't help you much on the points you raised at the moment, since fabric8 is a big project with many parts (and there are much smarter people which can help with console and CD issues).

And its summer, and some of us are on PTO right now. But if you bare with us, I'm pretty sure that we can fix this. @rawlingsj @jstrachan could you please have a look ?

Sorry for now, but please feel free to visit us also via IRC on #fabric8 on Freenode, where we hang out and do what we can.

**Comment by rawlingsj** [comment]
@rpots yeah everything @rhuss just said. Currently I'm on PTO until the 15th of this month but will have time to document the AWS experience and help with the issues you mentioned next week. Your help would definitely be appreciated and hopefully we can create the same OOTB experience for AWS and all Kubernetes based deployments.

**Comment by rawlingsj** [comment]
It's by no means polished but just to get us started I've created a getting started on AWS guide http://fabric8.io/guide/getStarted/aws.html

@rpots if you would like to add / update these docs you can submit a PR, the md docs can be found https://github.com/fabric8io/fabric8/blob/master/docs/getStarted/aws.md

FWIW after a digging a little this evening I found there's an env var `KUBE_ENABLE_INSECURE_REGISTRY` we can set before running the Kubernetes cluster up script.  This means the insecure registry flag is automatically added to the docker config of each node.  I've just tested it and it worked so hopefully that's a better solution to manually updating each node.

As for the other issues you mentioned, we're a little light on our feet this week too but will try to get them addressed asap.
